src/app/routes/register/register.component.html

```diff
@@ -53,23 +53,24 @@ <h2 class="mat-title" i18n="section heading create new goal">
 
         <mat-form-field appearance="outline" class="w-full">
           <mat-label>First name</mat-label>
-          <input type="text" matInput placeholder="First Name" formControlName="firstName">
+          <input type="text" matInput placeholder="First Name" formControlName="firstName" required>
```

Comment:
For reactive form, there is an option, Validators.required, if this is enabled, we can validate in template with .touched, .dirty... please use it, and remove required in all the places,